tools/monorepo-utils/src/workflow-profiler/commands/profile/index.ts

```diff
@@ -0,0 +1,54 @@
+/**
+ * External dependencies
+ */
+import { Command } from '@commander-js/extra-typings';
+
+/**
+ * Internal dependencies
+ */
+import {
+	getWorkflowRunData,
+	logWorkflowRunResults,
+	getWorkflowData,
+} from '../../lib';
+import { Logger } from '../../../core/logger';
+import config from '../../config';
+
+const program = new Command( 'profile' )
+	.description( 'Profile Github workflows' )
+	.argument( '<start>', 'Start date in YYYY-MM-DD format' )
+	.argument( '<end>', 'End date in YYYY-MM-DD format' )
+	.argument( '[ids...]', 'Workflow IDs. Defaults to required workflows' )
+	.option(
+		'-o --owner <owner>',
+		'Repository owner. Default: woocommerce',
+		'woocommerce'
+	)
+	.option(
+		'-n --name <name>',
+		'Repository name. Default: woocommerce',
+		'woocommerce'
+	)
+	.action( async ( start, end, ids, { owner, name } ) => {
+		const { requiredWorkflows } = config;
+		const workflowIds = ids.length ? ids : requiredWorkflows;
+
+		Promise.all(
```

Comment:
As you and @ObliviousHarmony have already discussed, the current approach uses lots of API calls, so we should definitely take some time to optimize it. For now though it works correctly, which to me is a valuable first iteration.